### ISSUE TITLE
not able to use external "vue" script

### ISSUE THREAD

**Issue by Urie96** [vuejs/vue-cli#6149]
### Version
5.0.0-alpha.0

### Reproduction link
[https://github.com/Urie96/vue-cli-issue](https://github.com/Urie96/vue-cli-issue)

### Environment info
webpack5 has new feature [externalsType](https://webpack.js.org/configuration/externals/#externalstype),  I try to extern vue by script, but i received a error in browser console when import `*.vue`,  it occur at last line  

```
exports.install = function (vue, browserify) {
  if (installed) { return }
  installed = true

  Vue = vue.__esModule ? vue.default : vue
  version = Vue.version.split('.').map(Number)
```

`Vue` is pending Promise, so version is undefined

### Steps to reproduce
- `vue create temp` 
- select vue 2 and disable all plugin
- `vi vue.config.js`
```js
module.exports = {
  configureWebpack: (config) => ({
    externalsType: 'script',
    externals: {
        vue: ['https://cdn.jsdelivr.net/npm/vue@2.6.12/dist/vue.min.js', 'Vue'],
    }})
}
```
- npm run serve
- open [chrome](http://localhost:8080)
- open <kbd>F12</kbd> and click `console`

### What is expected?
Be able to see vue.js page

### What is actually happening?
white page and chrome console throw error

**Comment by fangbinwei** [comment]
It seems it can be fixed in `vue-loader` by replacing [`api.install(require('vue')`](https://github.com/vuejs/vue-loader/blob/74020775b3e5a814c87d99381564e0d9e3d8d912/lib/codegen/hotReload.js#L19) with 
```js
import Vue from 'vue' 
api.install(Vue)
```
then webpack can know 'vue' is used and await Promise of script

**Comment by Urie96** [comment]
@fangbinwei It works fine if serve static files after `npm run build`. Is there something wrong in `vue-cli-service`?

**Comment by fangbinwei** [comment]
Some HMR codes use 'vue' by `require('vue')` which return Promise because of `externalsType`. HMR codes is injected only in dev mode, so it's no problem in production. @Urie96 

Using `externalsType: 'script'`,  `require('vue')` return Promise

```js
require('vue').then(Vue => {
  new Vue({})
})
```

if use vue like below,  webpack will transform codes and  handle the Promise automatically
```js
import Vue from 'vue'
new Vue({})
```

**Comment by Urie96** [comment]
@fangbinwei 谢谢解答，你上面讲的我配置不了哒，所以是要等待新的发布哇

**Comment by fangbinwei** [comment]
是的, 现在只能先这样
```js
/** @type {import('@vue/cli-service').ProjectOptions} */
module.exports = {
  configureWebpack: (config) => {
    if (process.env.NODE_ENV === 'production') {
      return {
        externalsType: 'script',
        externals: {
          vue: ['https://cdn.jsdelivr.net/npm/vue@2.6.12/dist/vue.min.js', 'Vue']
        }
      }
    }
  }
}
```

**Comment by aguegu** [comment]
I have a similar problem but not involve webpack5 yet.

With a freshly created project with @vue/cli 4.5.9, if I add 

```
module.exports = {
  chainWebpack: (config) => {
    config.externals({
      vue: 'Vue',      
    });
  },
};
```

in `vue.config.js`, and 

```
<script src="https://cdn.jsdelivr.net/npm/vuetify@2.3/dist/vuetify.min.js"></script>
```

in `index.html`.

And then I remove the local vue by `npm uninstall vue`, then I got `No loader specified` problem.

But it used to work fine with @vue/cli 4.4.6.

**Comment by fangbinwei** [comment]
@aguegu I can't reproduce this problem, maybe you can open a new issue and provide reproducible repo

**Comment by aguegu** [comment]
Thanks @fangbinwei , looks like you got a lot of open issues already. 

Here is the demo code without node_modules. 

[spa2.tar.gz](https://github.com/vuejs/vue-cli/files/5733675/spa2.tar.gz)

1. vue create spa2 (vue 2.0)
2. create `vue.config.js` with
```
module.exports = {
  chainWebpack: (config) => {
    config.externals({
      vue: 'Vue',      
    });
  },
};
```
3. Add 
```
<script src="https://cdn.jsdelivr.net/npm/vue@2.x/dist/vue.min.js"></script>
```
to `public/index.html`
4. npm uninstall vue
5. npm run serve

Somehow this time, the error is more detailed.

![Screenshot_2020-12-23_14-43-54](https://user-images.githubusercontent.com/1559791/102967624-51041900-452d-11eb-9469-e35a4d282552.png)

**Comment by fangbinwei** [comment]
@aguegu since `@vue/cli-service` will use different version of `vue-loader` according to the version of vue. If you uninstalled vue, `@vue/cli-service` couldn't apply `vue-loader` correctly, and webpack  failed to parse  `.vue` files. I don't recommend uninstalling vue. If you want to discuss more, you can open a new issue.

> vue3 and vue2 use different version of vue-loader

**Comment by aguegu** [comment]
@fangbinwei I do not want to uninstall vue that badly. What I want is to load vue.js from CDN. So what you are telling me is that from @vue/cli 4.5.9, vue.js can not be loaded from CDN, right?

**Comment by fangbinwei** [comment]
> vue.js can not be loaded from CDN, right?

No, use `externals` and do not uninstall vue, it won't be packaged into the bundle. You can compare the size of the bundle before and after external. Please do some tests by yourself. @aguegu

**Comment by aguegu** [comment]
@fangbinwei Oh, I tried and verified what you said. Thank you very much. I used to uninstall vue to make sure we get it from CDN. Since the dist file is impact. I am okay to leave it alone. :)
